### PR TITLE
Create revpi-v8_defconfig

### PR DESCRIPTION
**Pull request by nbuchwitz**
This PR creates `revpi-v8_defconfig` based on `bcm2711_defconfig`. All RevPi specific config options are applied in the second commit. The third commit which disabled KVM support, is a necessary fix without `PREEMPT_RT` cannot be choosen. For more details see the article linked in the commit message. Even if `PREEMPT_RT` is not functional with the second commit, I deliberately choose to set the option. This will allow us to revert  bae726be2e1936c252de30d3bf85f9c88210f66f when the problem is fixed upstream (still needed with Kernel 5.15.65 though: https://github.com/raspberrypi/linux/issues/5168#issuecomment-1241240866)

### CODE REVIEW

**Review by l1k — APPROVED** [review]
LGTM but should be reviewed by @iluminat23 as well before it gets merged.

**Comment by l1k** [comment]
Missing in 45c6675:
```
CONFIG_RTC_DRV_PCF2127=y
```

**Review by l1k — DISMISSED** [review]
_

**Comment by iluminat23** [comment]
Could you add a summary what is done in the log the 2nd commit? As this is essentially the squashed commits from `revpi-*_defconfig` you could add the squashed commit messages (without the `arm/configs/revpi-*` prefix) in there.

**Comment by iluminat23** [comment]
and use the same prefix as the other config changes: `arm64/configs/revpi-v8:`

**Comment by l1k** [comment]
> Could you add a summary what is done in the log the 2nd commit? As this is essentially the squashed commits from `revpi-*_defconfig` you could add the squashed commit messages (without the `arm/configs/revpi-*` prefix) in there.

Or better yet, convert the whole thing to config fragments as I recommended months ago to avoid missing settings. :-p

**Comment by nbuchwitz** [comment]
@l1k @iluminat23: I fixed the missing config option, added the squashed commit message and changed the prefix of all commit as suggested

**Comment by zhan-kunbus** [comment]
I compare the v8 and v7l, see additions of Android(CONFIG_ANDROID), oximeter (CONFIG_MAX30102), vhost net(CONFIG_VHOST_NET), etc.  I guess they are not needed by intention.

**Comment by nbuchwitz** [comment]
They are inherited from the `bcm2711_defconfig`. `CONFIG_MAX30102` is also present in newer 2711 arm defconfigs. I guess it is only absent in our v7l defconfig, because it was absent when we started our v7l defconfig:

```
grep -E 'CONFIG_ANDROID|CONFIG_MAX30102|CONFIG_VHOST_NET' arch/arm/configs/bcm2711_defconfig 
CONFIG_MAX30102=m
```

```
grep -E 'CONFIG_ANDROID|CONFIG_MAX30102|CONFIG_VHOST_NET' arch/arm64/configs/bcm2711_defconfig
CONFIG_VHOST_NET=m
CONFIG_MAX30102=m
CONFIG_ANDROID=y
CONFIG_ANDROID_BINDER_IPC=y
CONFIG_ANDROID_BINDERFS=y
```

**Comment by iluminat23** [comment]
1st commit message: " arm64/configs/revpi-v8: Create revpi-v8_defconfig from bcm2711_defconfig" => " arm64/configs: Create revpi-v8_defconfig from bcm2711_defconfig"
As there is no revpi-v8_defconfig before the commit.
2nd: commit " arm64/configs/revpi-v8_defconfig: Add RevPi specific configuration" => " arm64/configs/revpi-v8: Add RevPi specific configuration"
Just save some space in the head line of the commit message
3rd commit message: "arm64/configs/revpi-v8_defconfig: Disable CONFIG_KVM " => "arm64/configs/revpi-v8: Disable CONFIG_KVM"

**Comment by nbuchwitz** [comment]
@iluminat23 Makes sense. I amended the commits as suggested.

**Review by iluminat23 — COMMENTED** [review]
Looks good to me. @l1k @zhan-kunbus if you are OK with this as well you can approve this PR. I can't fully focus on this currently.

**Comment by zhan-kunbus** [comment]
> They are inherited from the `bcm2711_defconfig`. `CONFIG_MAX30102` is also present in newer 2711 arm defconfigs. I guess it is only absent in our v7l defconfig, because it was absent when we started our v7l defconfig:
> 
> ```
> grep -E 'CONFIG_ANDROID|CONFIG_MAX30102|CONFIG_VHOST_NET' arch/arm/configs/bcm2711_defconfig 
> CONFIG_MAX30102=m
> ```
> 
> ```
> grep -E 'CONFIG_ANDROID|CONFIG_MAX30102|CONFIG_VHOST_NET' arch/arm64/configs/bcm2711_defconfig
> CONFIG_VHOST_NET=m
> CONFIG_MAX30102=m
> CONFIG_ANDROID=y
> CONFIG_ANDROID_BINDER_IPC=y
> CONFIG_ANDROID_BINDERFS=y
> ```

I meant it seems they are not needed on our devices, e.g. for CONFIG_MAX30102, no hardware on our devices, and for Android, it isn't supported by our devices as I think, etc.  So the modules should not be part of our kernel, isn't it?   they enlarge the size of kernel and enlengthen the booting time when be loaded by default.

**Comment by nbuchwitz** [comment]
Valid point. One could argue that we have lots of irrelevant options enabled in our already existing defconfigs. But we don't have the options enabled in our revpi-v7l, so I would be ok with two additional commits which remove CONFIG_MAX30102 as well as CONFIG_ANDROID (esp. after some read-up here https://lwn.net/Articles/899743/).

@l1k @linosanfilippo-kunbus @iluminat23 any thoughts on this?

**Comment by l1k** [comment]
Go for it. Creating a custom config which is geared specifically towards RevPi products was precisely the motivation for avoiding Kconfig fragments (which I would have preferred).

It could be argued that such cleanups aren't necessarily part of 64-bit enablement, but whatever.

Enabling `CONFIG_MAX30102` doesn't increase boot time, but build time.

**Comment by nbuchwitz** [comment]
@zhan-kunbus, please check the latest two commits and let me know if everything is ok now.

**Comment by iluminat23** [comment]
I'm in favor to keep the kernels the same from point of drivers which they include. So removing CONFIG_MAX30102 from v8 or adding to the other!

**Comment by iluminat23** [comment]
Ohhh and yes the ANDROID specific config options should be dropped

**Comment by zhan-kunbus** [comment]
> Valid point. One could argue that we have lots of irrelevant options enabled in our already existing defconfigs. But we don't have the options enabled in our revpi-v7l, so I would be ok with two additional commits which remove CONFIG_MAX30102 as well as CONFIG_ANDROID (esp. after some read-up here https://lwn.net/Articles/899743/).
> 
> @l1k @linosanfilippo-kunbus @iluminat23 any thoughts on this?

maybe we can create a ticket to remove the irrelevant options for better performance/size/failure.. like CONFIG_PCI(_MSI) in v7l, I guess we have no pci and no the possibility.

**Comment by nbuchwitz** [comment]
Good idea. It would be also great, to have one, unified defconf for v7(l). This would allow us to simplify our kernel packages in the fure (only one arm32 and one arm64 build). But this is way out of scope for this PR.  Let's continue this discussion internally.

TL;DR; Is there anything missing for this v8 defconfig?

**Review by zhan-kunbus — APPROVED** [review]
